### PR TITLE
CI Publishing

### PR DESCRIPTION
**Pull request by akdetrick**
**DO NOT MERGE**
This code is probably broken. Submitting a PR as a starting point to sort out jenkins config and resolve open questions around CI

---

### For the engineer...
1. An engineer branches `new-feature` from `master` to make a code change. For this example, let's assume the change is a _minor_ change (new feature, backwards compatible).
2. The engineer updates `CHANGELOG.md` to add notes for their _minor_ version.
3. The engineer updates `Makefile` to bump the _minor_ version.
4. The engineer submits a pull request.

### For Jenkins, running on the PR
1. Run _install_, _test_, and _build_ steps of the pipeline
2. Publish a beta version of fds packages to npm
3. Git tag the tip of the branch (`major.minor.(build number)-beta`) and push tag to origin

### For Jenkins, running on a merge to `master`
1. Run _install_, _test_, and _build_ steps of the pipeline
2. Publish a stable version of fds packages to npm
3. Git tag the tip of the branch (`major.minor.(build number)`) and push tag to origin

---

### Environment variables
env var                     | set by      | description
------------------- | --------- | ---------------
`BUILD_NUBMER`  | Jenkins   | Jenkins build number
`NPM_TAG`             | us            | [npm dist tag](https://docs.npmjs.com/cli/dist-tag#purpose) - tells npm if `latest` or `beta`
`GIT_TAG`                | us            | version number for git tag _and_ npm version number

Basically, Jenkins can call `make version` to get the correct version number for the release. We determine if this is `beta` or not based on `BRANCH_NAME`. 

Once the above environment variables are set according to these rules, we run a `lerna` command to publish to npm with the given version number and NPM dist.

Once the `lerna` publish step completes, Jenkins should git tag the tip of the branch with that version number and dist tag, and push the tag to origin.

### CODE REVIEW

**Comment by akdetrick** [comment]
done in another branch
